Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphListener/MicrosoftGraphListener.py

```diff
@@ -160,20 +71,37 @@ def fetch_incidents(self, last_run):
             last_fetch, _ = parse_date_range(self._first_fetch_interval, date_format=DATE_FORMAT, utc=True)
             demisto.info(f"initialize fetch and pull emails from date :{last_fetch}")
 
-        fetched_emails, fetched_emails_ids = self._fetch_last_emails(folder_id=folder_id, last_fetch=last_fetch,
-                                                                     exclude_ids=exclude_ids)
+        fetched_emails, exclude_ids = self._fetch_last_emails(folder_id=folder_id, last_fetch=last_fetch,
+                                                              exclude_ids=exclude_ids)
+
         incidents = [self._parse_email_as_incident(email, True) for email in fetched_emails]
+
         next_run_time = self._get_next_run_time(fetched_emails, last_fetch)
+
         next_run = {
             'LAST_RUN_TIME': next_run_time,
-            'LAST_RUN_IDS': fetched_emails_ids,
+            'LAST_RUN_IDS': exclude_ids,
             'LAST_RUN_FOLDER_ID': folder_id,
             'LAST_RUN_FOLDER_PATH': self._folder_to_fetch
         }
-        demisto.info(f"fetched {len(incidents)} incidents")
+
         demisto.debug(f'MicrosoftGraphMail - Next run after incidents fetching: {json.dumps(next_run)}')
         demisto.debug(f"MicrosoftGraphMail - Number of incidents before filtering: {len(fetched_emails)}")
         demisto.debug(f"MicrosoftGraphMail - Number of incidents after filtering: {len(incidents)}")
+        demisto.debug(f"MicrosoftGraphMail - Number of incidents skipped: {len(fetched_emails)-len(incidents)}")
+        for incident in incidents:  # remove the ID from the incidents, they are used only for look-back.
+            incident.pop('ID', None)
```

Comment:
As discussed in the demo session today, this is here due to the difference in limits and duplicate handling between the two integrations. Consideration should be given in the future to aligning the top-level fetch commands in each of these integrations as I don't believe the extra code in the MicrosoftGraphMail integration is necessary and if removed would remove the requirements for the superfluous pop in this integration.

Reply:
So please add a code comment explaining the purpose of the pop Here. This will help future programmers understand why it is needed when they encounter this section of code.

Reply:
done in PR 29839